parsers/verilog.c

```diff
@@ -46,9 +46,13 @@
  * If you will separate the definitions for the parsers, you must revise the
  * code related to the symbol table. */
 typedef enum {
-	K_IGNORE = -2,
-	K_UNDEFINED,
-	K_CONSTANT,
+	/* parser private items */
+	K_IGNORE = -16,
+	K_DEFINE,
```

Comment:
s it the best way to tag "foo" of "`define foo ..." with constant kind?
Introducing "d/definition" kind and tagging the "foo" with the definition kind is an alternative way.

Using the unified kind "constant" in many areas reduces the information that client tools get from a tags file.
In my experience, it is better to tag different things with different kinds. If ctags hides the differences, and just reports various language objects as "constant," in some cases, a client tool must parse the raw .sv source file for distinguishing whether a given tag is "`define"'ed or not.

IMHO, ctags should not be too smart. Instead, ctags should be stupid; it should report what it sees in source input files as-is to client tools that will do something smart.

Reply:
Sorry, what I wrote here is pointless; introducing the d/definition kind breaks compatibility.
So introducing the kind is not so easy.

Reply:
You are right.
And I also agree with you that `define should have its own kind if we are allowed some incompatibility.